packages/app/src/components/WalletStatusBar.vue

```diff
@@ -0,0 +1,88 @@
+<template>
+  <div class="wallet-status-bar">
+    <NetworkIndicator :currentNetwork="currentNetwork" :isWrongNetwork="isWrongNetwork" />
+    <ConnectMetamaskButton />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, onMounted, onUnmounted, ref, watchEffect } from "vue";
+
+import ConnectMetamaskButton from "@/components/ConnectMetamaskButton.vue";
+import NetworkIndicator from "@/components/NetworkIndicator.vue";
+
+import useContext from "@/composables/useContext";
+import useLogin from "@/composables/useLogin";
+import { default as useWallet } from "@/composables/useWallet";
+
+const context = useContext();
+const { logout } = useLogin(context);
+const currentNetwork = computed(() => context.currentNetwork.value);
+
+const { address, getEthereumProvider, isReady } = useWallet({
+  ...context,
+  currentNetwork: computed(() => ({
+    explorerUrl: context.currentNetwork.value.rpcUrl,
+    chainName: context.currentNetwork.value.l2NetworkName,
+    l1ChainId: null as unknown as number,
```

Comment:
Is it ok to have the contstant null here?

Also, typescript it's a little bit special with the `as`. Even when this type, I feel that sending "null" as a number might cause issues down the road.

Reply:
Yes I agree, I didn't want to make it optional in the type so maybe passing a value that makes sense here and is not bypassing typescript type checking could be a better idea.